### PR TITLE
fix: store asset failure report supports exception without logs

### PR DESCRIPTION
**Pull request by guilhem-barthes**
## Description

Fixes Fl-1689

## Checklist

- [x] [changelog](../CHANGELOG.md) was updated with notable changes
- [ ] documentation was updated

### CODE REVIEW

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

File: charts/substra-backend/changes/970.fixed [hunk 1]
```diff
@@ -0,0 +1 @@
+Celery was failing silently when a task called `FailableTask.on_failure` if the task didn't have a `logs` attribute (now return internal error)
```
Comment (SdgJlbl) [inline charts/substra-backend/changes/970.fixed:1]: I don't think that the right `changes` folder :)

File: backend/builder/exceptions.py [hunk 2]
```diff
@@ -14,7 +13,7 @@ class PodTimeoutError(Exception):
     pass
 
 
-class BuildRetryError(_ComputeTaskError, CeleryRetryError):
+class _BuildError(RuntimeError):
```
Comment (SdgJlbl) [inline backend/builder/exceptions.py:16]: So if I understand correctly, `BuildError` will never share its logs? 🤔

**Review by SdgJlbl — APPROVED** [review]
🙏

**Pull request merged**